The job **`x86_64-gnu-llvm-18`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/11184957499/job/31096849560) [(plain)](https://github.com/rust-lang/rust/commit/1556ac851e0b1c7dd84e8d885518aa36ee57e819/checks/31096849560/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
------
 > importing cache manifest from ghcr.io/rust-lang/rust-ci-cache:20950f2ccee3dff53a038adf5c1cf05231c0b30772617126a5f6478a66316a29cba9aead69f7bb0004886d32c1f8e6287542cf6d25130711c82d16a66201d4fe:
------
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-18]
---
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-18', '--enable-llvm-link-shared', '--set', 'rust.randomize-layout=true', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'rust.lld=false', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-18/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.randomize-layout := True
configure: rust.thin-lto-import-instr-limit := 10
---
running 17724 tests
iiiiii...................................................i................ii.ii.......
[ui] tests/ui/abi/riscv32e-registers.rs#riscv32em ... F

[ui] tests/ui/abi/riscv32e-registers.rs#riscv32emc ... F

[ui] tests/ui/abi/riscv32e-registers.rs#riscv32e ... F
......iii............................i.i.i.............................................   176/17724
...........................................................iiiiiiiiiiiiii.....i.........   264/17724
---
1 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:42:11
+   --> $DIR/riscv32e-registers.rs:44:11
3    |
4 LL |     asm!("li x16, 0");

11    |        ^
12 
13 error: invalid operand for instruction
13 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:45:11
+   --> $DIR/riscv32e-registers.rs:47:11
15    |
16 LL |     asm!("li x17, 0");

23    |        ^
24 
25 error: invalid operand for instruction
25 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:48:11
+   --> $DIR/riscv32e-registers.rs:50:11
27    |
28 LL |     asm!("li x18, 0");

35    |        ^
36 
37 error: invalid operand for instruction
37 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:51:11
+   --> $DIR/riscv32e-registers.rs:53:11
39    |
40 LL |     asm!("li x19, 0");

47    |        ^
48 
49 error: invalid operand for instruction
49 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:54:11
+   --> $DIR/riscv32e-registers.rs:56:11
51    |
52 LL |     asm!("li x20, 0");

59    |        ^
60 
61 error: invalid operand for instruction
61 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:57:11
+   --> $DIR/riscv32e-registers.rs:59:11
63    |
64 LL |     asm!("li x21, 0");

71    |        ^
72 
73 error: invalid operand for instruction
73 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:60:11
+   --> $DIR/riscv32e-registers.rs:62:11
75    |
76 LL |     asm!("li x22, 0");

83    |        ^
84 
85 error: invalid operand for instruction
85 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:63:11
+   --> $DIR/riscv32e-registers.rs:65:11
87    |
88 LL |     asm!("li x23, 0");

95    |        ^
96 
97 error: invalid operand for instruction
97 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:66:11
+   --> $DIR/riscv32e-registers.rs:68:11
99    |
100 LL |     asm!("li x24, 0");

107    |        ^
108 
109 error: invalid operand for instruction
109 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:69:11
+   --> $DIR/riscv32e-registers.rs:71:11
111    |
112 LL |     asm!("li x25, 0");

119    |        ^
120 
121 error: invalid operand for instruction
121 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:72:11
+   --> $DIR/riscv32e-registers.rs:74:11
123    |
124 LL |     asm!("li x26, 0");

131    |        ^
132 
133 error: invalid operand for instruction
133 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:75:11
+   --> $DIR/riscv32e-registers.rs:77:11
135    |
136 LL |     asm!("li x27, 0");

143    |        ^
144 
145 error: invalid operand for instruction
145 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:78:11
+   --> $DIR/riscv32e-registers.rs:80:11
147    |
148 LL |     asm!("li x28, 0");

155    |        ^
156 
157 error: invalid operand for instruction
157 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:81:11
+   --> $DIR/riscv32e-registers.rs:83:11
159    |
160 LL |     asm!("li x29, 0");

167    |        ^
168 
169 error: invalid operand for instruction
169 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:84:11
+   --> $DIR/riscv32e-registers.rs:86:11
171    |
172 LL |     asm!("li x30, 0");

179    |        ^
180 
181 error: invalid operand for instruction
181 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:87:11
+   --> $DIR/riscv32e-registers.rs:89:11
183    |
184 LL |     asm!("li x31, 0");


The actual stderr differed from the expected stderr.
The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/abi/riscv32e-registers.riscv32em/riscv32e-registers.riscv32em.stderr
To only update this specific test, also pass `--test-args abi/riscv32e-registers.rs`


error in revision `riscv32em`: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/abi/riscv32e-registers.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--cfg" "riscv32em" "--check-cfg" "cfg(FALSE,riscv32e,riscv32em,riscv32emc)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/abi/riscv32e-registers.riscv32em" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/abi/riscv32e-registers.riscv32em/auxiliary" "--crate-type=rlib" "--target=riscv32em-unknown-none-elf"
--- stderr -------------------------------
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:44:11
   |
   |
LL |     asm!("li x16, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x16, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:47:11
   |
LL |     asm!("li x17, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x17, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:50:11
   |
LL |     asm!("li x18, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x18, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:53:11
   |
LL |     asm!("li x19, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x19, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:56:11
   |
LL |     asm!("li x20, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x20, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:59:11
   |
LL |     asm!("li x21, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x21, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:62:11
   |
LL |     asm!("li x22, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x22, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:65:11
   |
LL |     asm!("li x23, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x23, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:68:11
   |
LL |     asm!("li x24, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x24, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:71:11
   |
LL |     asm!("li x25, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x25, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:74:11
   |
LL |     asm!("li x26, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x26, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:77:11
   |
LL |     asm!("li x27, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x27, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:80:11
   |
LL |     asm!("li x28, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x28, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:83:11
   |
LL |     asm!("li x29, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x29, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:86:11
   |
LL |     asm!("li x30, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x30, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:89:11
   |
LL |     asm!("li x31, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x31, 0
   |        ^

error: aborting due to 16 previous errors
error: aborting due to 16 previous errors
------------------------------------------


---- [ui] tests/ui/abi/riscv32e-registers.rs#riscv32emc stdout ----

1 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:42:11
+   --> $DIR/riscv32e-registers.rs:44:11
+   --> $DIR/riscv32e-registers.rs:44:11
3    |
4 LL |     asm!("li x16, 0");

11    |        ^
12 
13 error: invalid operand for instruction
13 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:45:11
+   --> $DIR/riscv32e-registers.rs:47:11
15    |
16 LL |     asm!("li x17, 0");

23    |        ^
24 
25 error: invalid operand for instruction
25 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:48:11
+   --> $DIR/riscv32e-registers.rs:50:11
27    |
28 LL |     asm!("li x18, 0");

35    |        ^
36 
37 error: invalid operand for instruction
37 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:51:11
+   --> $DIR/riscv32e-registers.rs:53:11
39    |
40 LL |     asm!("li x19, 0");

47    |        ^
48 
49 error: invalid operand for instruction
49 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:54:11
+   --> $DIR/riscv32e-registers.rs:56:11
51    |
52 LL |     asm!("li x20, 0");

59    |        ^
60 
61 error: invalid operand for instruction
61 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:57:11
+   --> $DIR/riscv32e-registers.rs:59:11
63    |
64 LL |     asm!("li x21, 0");

71    |        ^
72 
73 error: invalid operand for instruction
73 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:60:11
+   --> $DIR/riscv32e-registers.rs:62:11
75    |
76 LL |     asm!("li x22, 0");

83    |        ^
84 
85 error: invalid operand for instruction
85 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:63:11
+   --> $DIR/riscv32e-registers.rs:65:11
87    |
88 LL |     asm!("li x23, 0");

95    |        ^
96 
97 error: invalid operand for instruction
97 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:66:11
+   --> $DIR/riscv32e-registers.rs:68:11
99    |
100 LL |     asm!("li x24, 0");

107    |        ^
108 
109 error: invalid operand for instruction
109 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:69:11
+   --> $DIR/riscv32e-registers.rs:71:11
111    |
112 LL |     asm!("li x25, 0");

119    |        ^
120 
121 error: invalid operand for instruction
121 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:72:11
+   --> $DIR/riscv32e-registers.rs:74:11
123    |
124 LL |     asm!("li x26, 0");

131    |        ^
132 
133 error: invalid operand for instruction
133 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:75:11
+   --> $DIR/riscv32e-registers.rs:77:11
135    |
136 LL |     asm!("li x27, 0");

143    |        ^
144 
145 error: invalid operand for instruction
145 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:78:11
+   --> $DIR/riscv32e-registers.rs:80:11
147    |
148 LL |     asm!("li x28, 0");

155    |        ^
156 
157 error: invalid operand for instruction
157 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:81:11
+   --> $DIR/riscv32e-registers.rs:83:11
159    |
160 LL |     asm!("li x29, 0");

167    |        ^
168 
169 error: invalid operand for instruction
169 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:84:11
+   --> $DIR/riscv32e-registers.rs:86:11
171    |
172 LL |     asm!("li x30, 0");

179    |        ^
180 
181 error: invalid operand for instruction
181 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:87:11
+   --> $DIR/riscv32e-registers.rs:89:11
183    |
184 LL |     asm!("li x31, 0");


The actual stderr differed from the expected stderr.
The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/abi/riscv32e-registers.riscv32emc/riscv32e-registers.riscv32emc.stderr
To only update this specific test, also pass `--test-args abi/riscv32e-registers.rs`


---
1 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:42:11
+   --> $DIR/riscv32e-registers.rs:44:11
3    |
4 LL |     asm!("li x16, 0");

11    |        ^
12 
13 error: invalid operand for instruction
13 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:45:11
+   --> $DIR/riscv32e-registers.rs:47:11
15    |
16 LL |     asm!("li x17, 0");

23    |        ^
24 
25 error: invalid operand for instruction
25 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:48:11
+   --> $DIR/riscv32e-registers.rs:50:11
27    |
28 LL |     asm!("li x18, 0");

35    |        ^
36 
37 error: invalid operand for instruction
37 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:51:11
+   --> $DIR/riscv32e-registers.rs:53:11
39    |
40 LL |     asm!("li x19, 0");

47    |        ^
48 
49 error: invalid operand for instruction
49 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:54:11
+   --> $DIR/riscv32e-registers.rs:56:11
51    |
52 LL |     asm!("li x20, 0");

59    |        ^
60 
61 error: invalid operand for instruction
61 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:57:11
+   --> $DIR/riscv32e-registers.rs:59:11
63    |
64 LL |     asm!("li x21, 0");

71    |        ^
72 
73 error: invalid operand for instruction
73 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:60:11
+   --> $DIR/riscv32e-registers.rs:62:11
75    |
76 LL |     asm!("li x22, 0");

83    |        ^
84 
85 error: invalid operand for instruction
85 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:63:11
+   --> $DIR/riscv32e-registers.rs:65:11
87    |
88 LL |     asm!("li x23, 0");

95    |        ^
96 
97 error: invalid operand for instruction
97 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:66:11
+   --> $DIR/riscv32e-registers.rs:68:11
99    |
100 LL |     asm!("li x24, 0");

107    |        ^
108 
109 error: invalid operand for instruction
109 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:69:11
+   --> $DIR/riscv32e-registers.rs:71:11
111    |
112 LL |     asm!("li x25, 0");

119    |        ^
120 
121 error: invalid operand for instruction
121 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:72:11
+   --> $DIR/riscv32e-registers.rs:74:11
123    |
124 LL |     asm!("li x26, 0");

131    |        ^
132 
133 error: invalid operand for instruction
133 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:75:11
+   --> $DIR/riscv32e-registers.rs:77:11
135    |
136 LL |     asm!("li x27, 0");

143    |        ^
144 
145 error: invalid operand for instruction
145 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:78:11
+   --> $DIR/riscv32e-registers.rs:80:11
147    |
148 LL |     asm!("li x28, 0");

155    |        ^
156 
157 error: invalid operand for instruction
157 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:81:11
+   --> $DIR/riscv32e-registers.rs:83:11
159    |
160 LL |     asm!("li x29, 0");

167    |        ^
168 
169 error: invalid operand for instruction
169 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:84:11
+   --> $DIR/riscv32e-registers.rs:86:11
171    |
172 LL |     asm!("li x30, 0");

179    |        ^
180 
181 error: invalid operand for instruction
181 error: invalid operand for instruction
-   --> $DIR/riscv32e-registers.rs:87:11
+   --> $DIR/riscv32e-registers.rs:89:11
183    |
184 LL |     asm!("li x31, 0");


The actual stderr differed from the expected stderr.
The actual stderr differed from the expected stderr.
Actual stderr saved to /checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/abi/riscv32e-registers.riscv32e/riscv32e-registers.riscv32e.stderr
To only update this specific test, also pass `--test-args abi/riscv32e-registers.rs`


error in revision `riscv32e`: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/abi/riscv32e-registers.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--cfg" "riscv32e" "--check-cfg" "cfg(FALSE,riscv32e,riscv32em,riscv32emc)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/abi/riscv32e-registers.riscv32e" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/abi/riscv32e-registers.riscv32e/auxiliary" "--crate-type=rlib" "--target=riscv32e-unknown-none-elf"
--- stderr -------------------------------
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:44:11
   |
   |
LL |     asm!("li x16, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x16, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:47:11
   |
LL |     asm!("li x17, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x17, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:50:11
   |
LL |     asm!("li x18, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x18, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:53:11
   |
LL |     asm!("li x19, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x19, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:56:11
   |
LL |     asm!("li x20, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x20, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:59:11
   |
LL |     asm!("li x21, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x21, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:62:11
   |
LL |     asm!("li x22, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x22, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:65:11
   |
LL |     asm!("li x23, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x23, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:68:11
   |
LL |     asm!("li x24, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x24, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:71:11
   |
LL |     asm!("li x25, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x25, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:74:11
   |
LL |     asm!("li x26, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x26, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:77:11
   |
LL |     asm!("li x27, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x27, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:80:11
   |
LL |     asm!("li x28, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x28, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:83:11
   |
LL |     asm!("li x29, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x29, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:86:11
   |
LL |     asm!("li x30, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x30, 0
   |        ^

error: invalid operand for instruction
error: invalid operand for instruction
##[error]  --> /checkout/tests/ui/abi/riscv32e-registers.rs:89:11
   |
LL |     asm!("li x31, 0");
   |
note: instantiated into assembly here
note: instantiated into assembly here
  --> <inline asm>:1:5
LL |     li x31, 0
   |        ^

error: aborting due to 16 previous errors
```

</details>

--- a/compiler/rustc_target/src/spec/mod.rs
+++ b/compiler/rustc_target/src/spec/mod.rs
@@ -1841,6 +1841,10 @@ supported_targets! {
     ("riscv32imac-esp-espidf", riscv32imac_esp_espidf),
     ("riscv32imafc-esp-espidf", riscv32imafc_esp_espidf),
 
+    ("riscv32e-unknown-none-elf", riscv32e_unknown_none_elf),
+    ("riscv32em-unknown-none-elf", riscv32em_unknown_none_elf),
+    ("riscv32emc-unknown-none-elf", riscv32emc_unknown_none_elf),
+
     ("riscv32imac-unknown-none-elf", riscv32imac_unknown_none_elf),
     ("riscv32imafc-unknown-none-elf", riscv32imafc_unknown_none_elf),
     ("riscv32imac-unknown-xous-elf", riscv32imac_unknown_xous_elf),

--- a/compiler/rustc_target/src/spec/targets/riscv32e_unknown_none_elf.rs
+++ b/compiler/rustc_target/src/spec/targets/riscv32e_unknown_none_elf.rs
@@ -1,11 +1,17 @@
 use crate::spec::{Cc, LinkerFlavor, Lld, PanicStrategy, RelocModel, Target, TargetOptions};
 
-pub fn target() -> Target {
+pub(crate) fn target() -> Target {
     Target {
         // The below `data_layout` is explicitly specified by the ilp32e ABI in LLVM. See also
         // `options.llvm_abiname`.
         data_layout: "e-m:e-p:32:32-i64:64-n32-S32".into(),
         llvm_target: "riscv32".into(),
+        metadata: crate::spec::TargetMetadata {
+            description: Some("Bare RISC-V (RV32E ISA)".into()),
+            tier: Some(3),
+            host_tools: Some(false),
+            std: Some(false),
+        },
         pointer_width: 32,
         arch: "riscv32".into(),
 

--- a/compiler/rustc_target/src/spec/targets/riscv32e_unknown_none_elf.rs
+++ b/compiler/rustc_target/src/spec/targets/riscv32e_unknown_none_elf.rs
@@ -1,0 +1,25 @@
+use crate::spec::{Cc, LinkerFlavor, Lld, PanicStrategy, RelocModel, Target, TargetOptions};
+
+pub fn target() -> Target {
+    Target {
+        data_layout: "e-m:e-p:32:32-i64:64-n32-S32".into(),
+        llvm_target: "riscv32".into(),
+        pointer_width: 32,
+        arch: "riscv32".into(),
+
+        options: TargetOptions {
+            linker_flavor: LinkerFlavor::Gnu(Cc::No, Lld::Yes),
+            linker: Some("rust-lld".into()),
+            cpu: "generic-rv32".into(),
+            llvm_abiname: "ilp32e".into(),
+            max_atomic_width: Some(32),
+            atomic_cas: false,
+            features: "+e,+forced-atomics".into(),
+            panic_strategy: PanicStrategy::Abort,
+            relocation_model: RelocModel::Static,
+            emit_debug_gdb_scripts: false,
+            eh_frame_header: false,
+            ..Default::default()
+        },
+    }
+}

--- a/compiler/rustc_target/src/spec/targets/riscv32em_unknown_none_elf.rs
+++ b/compiler/rustc_target/src/spec/targets/riscv32em_unknown_none_elf.rs
@@ -1,0 +1,25 @@
+use crate::spec::{Cc, LinkerFlavor, Lld, PanicStrategy, RelocModel, Target, TargetOptions};
+
+pub fn target() -> Target {
+    Target {
+        data_layout: "e-m:e-p:32:32-i64:64-n32-S32".into(),
+        llvm_target: "riscv32".into(),
+        pointer_width: 32,
+        arch: "riscv32".into(),
+
+        options: TargetOptions {
+            linker_flavor: LinkerFlavor::Gnu(Cc::No, Lld::Yes),
+            linker: Some("rust-lld".into()),
+            cpu: "generic-rv32".into(),
+            llvm_abiname: "ilp32e".into(),
+            max_atomic_width: Some(32),
+            atomic_cas: false,
+            features: "+e,+m,+forced-atomics".into(),
+            panic_strategy: PanicStrategy::Abort,
+            relocation_model: RelocModel::Static,
+            emit_debug_gdb_scripts: false,
+            eh_frame_header: false,
+            ..Default::default()
+        },
+    }
+}

--- a/compiler/rustc_target/src/spec/targets/riscv32em_unknown_none_elf.rs
+++ b/compiler/rustc_target/src/spec/targets/riscv32em_unknown_none_elf.rs
@@ -1,11 +1,17 @@
 use crate::spec::{Cc, LinkerFlavor, Lld, PanicStrategy, RelocModel, Target, TargetOptions};
 
-pub fn target() -> Target {
+pub(crate) fn target() -> Target {
     Target {
         // The below `data_layout` is explicitly specified by the ilp32e ABI in LLVM. See also
         // `options.llvm_abiname`.
         data_layout: "e-m:e-p:32:32-i64:64-n32-S32".into(),
         llvm_target: "riscv32".into(),
+        metadata: crate::spec::TargetMetadata {
+            description: Some("Bare RISC-V (RV32EM ISA)".into()),
+            tier: Some(3),
+            host_tools: Some(false),
+            std: Some(false),
+        },
         pointer_width: 32,
         arch: "riscv32".into(),
 

--- a/compiler/rustc_target/src/spec/targets/riscv32emc_unknown_none_elf.rs
+++ b/compiler/rustc_target/src/spec/targets/riscv32emc_unknown_none_elf.rs
@@ -1,11 +1,17 @@
 use crate::spec::{Cc, LinkerFlavor, Lld, PanicStrategy, RelocModel, Target, TargetOptions};
 
-pub fn target() -> Target {
+pub(crate) fn target() -> Target {
     Target {
         // The below `data_layout` is explicitly specified by the ilp32e ABI in LLVM. See also
         // `options.llvm_abiname`.
         data_layout: "e-m:e-p:32:32-i64:64-n32-S32".into(),
         llvm_target: "riscv32".into(),
+        metadata: crate::spec::TargetMetadata {
+            description: Some("Bare RISC-V (RV32EMC ISA)".into()),
+            tier: Some(3),
+            host_tools: Some(false),
+            std: Some(false),
+        },
         pointer_width: 32,
         arch: "riscv32".into(),
 

--- a/compiler/rustc_target/src/spec/targets/riscv32emc_unknown_none_elf.rs
+++ b/compiler/rustc_target/src/spec/targets/riscv32emc_unknown_none_elf.rs
@@ -1,0 +1,25 @@
+use crate::spec::{Cc, LinkerFlavor, Lld, PanicStrategy, RelocModel, Target, TargetOptions};
+
+pub fn target() -> Target {
+    Target {
+        data_layout: "e-m:e-p:32:32-i64:64-n32-S32".into(),
+        llvm_target: "riscv32".into(),
+        pointer_width: 32,
+        arch: "riscv32".into(),
+
+        options: TargetOptions {
+            linker_flavor: LinkerFlavor::Gnu(Cc::No, Lld::Yes),
+            linker: Some("rust-lld".into()),
+            cpu: "generic-rv32".into(),
+            llvm_abiname: "ilp32e".into(),
+            max_atomic_width: Some(32),
+            atomic_cas: false,
+            features: "+e,+m,+c,+forced-atomics".into(),
+            panic_strategy: PanicStrategy::Abort,
+            relocation_model: RelocModel::Static,
+            emit_debug_gdb_scripts: false,
+            eh_frame_header: false,
+            ..Default::default()
+        },
+    }
+}

--- a/src/bootstrap/src/core/sanity.rs
+++ b/src/bootstrap/src/core/sanity.rs
@@ -37,6 +37,9 @@ pub struct Finder {
 const STAGE0_MISSING_TARGETS: &[&str] = &[
     // just a dummy comment so the list doesn't get onelined
     "armv7-rtems-eabihf",
+    "riscv32e-unknown-none-elf",
+    "riscv32em-unknown-none-elf",
+    "riscv32emc-unknown-none-elf",
 ];
 
 /// Minimum version threshold for libstdc++ required when using prebuilt LLVM

